### PR TITLE
Set folder path recording UI added

### PR DESCRIPTION
**Pull request by salmanmkc**
Hey I've created an additional section for setting a path for recording output.

It does this:
- Connecting to the server will load the current file path set for recording
- Browse button lets the user choose a folder for the recordings to be saved
- They can also manually type in the folder path and use the set path button

Tested this on Windows 10

Change is at the bottom of the Window

Before:
no folder path selection

After:
Folder path selection

### CODE REVIEW

File: TestClient/MainWindow.cs [hunk 1]
```diff
@@ -396,5 +401,21 @@ private void btnAdvanced_Click(object sender, EventArgs e)
...
         }
+
+        private void browseButton_Click(object sender, EventArgs e)
+        {
+            DialogResult result = this.folderBrowserDialog1.ShowDialog();
+            if (result == DialogResult.OK)
+            {
+                string selectedPath = this.folderBrowserDialog1.SelectedPath;
+                folderpath.Text = selectedPath;
+                _obs.SetRecordingFolder(selectedPath);
```
Comment (BarRaider) [inline TestClient/MainWindow.cs]: Redundant. Should only be in the "Set Path" button

File: TestClient/MainWindow.cs [hunk 2]
```diff
@@ -226,11 +228,12 @@ private void onStreamData(OBSWebsocket sender, StreamStatus data)
 
         private void btnConnect_Click(object sender, EventArgs e)
         {
-            if(!_obs.IsConnected)
+            if (!_obs.IsConnected)
             {
                 try
                 {
                     _obs.Connect(txtServerIP.Text, txtServerPassword.Text);
+                    folderpath.Text = _obs.GetRecordingFolder().ToString();
```
Comment (BarRaider) [inline TestClient/MainWindow.cs]: Should move to OnConnect like the rest of the settings that are populated when connected

**Review by BarRaider — CHANGES_REQUESTED** [review]

File: TestClient/MainWindow.cs (hunk 1 above)
Comment (salmanmkc) [reply TestClient/MainWindow.cs]: I thought that if they choose a folder with the picker that it's good to not need to set the path with the button after they've selected it through the folder picker

File: TestClient/MainWindow.cs (hunk 2 above)
Comment (salmanmkc) [reply TestClient/MainWindow.cs]: ah nice, didn't realise it was there, thanks

File: TestClient/MainWindow.cs (hunk 1 above)
Comment (BarRaider) [reply TestClient/MainWindow.cs]: That is confusing. And forcing a change a user may not want.

Comment (salmanmkc) [reply TestClient/MainWindow.cs]: understood, updated now to just update the textbox

**Pull request merged**